posts/the-game-of-vue-migrations.md

```diff
@@ -0,0 +1,158 @@
+
+One last tip before bringing up the code coverage, add this command to the `"scripts"` section of the `package.json` to run the tests continuously:
+
+```json
+"test:unit:watch": "vue-cli-service test:unit --watch --coverage"
+```
+
+With enough tests to bring the coverage reports up to compliant levels (generally think >80% but be sure to hit any edge cases!), the console output may look more like this:
+
+![Unit Testing Output with Full Code Coverage](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2021/10/VueUnitTestingOutputFullCodeCoverage.png)
+
+### Awareness of Where We are Headed
+
+Thinking ahead while writing the tests to the TypeScript and refactoring steps, here are a few opportunities for improvement:
+
+1. The default "`grid`" value of "`[[false]]`" within "`data()`" is an incorrect placeholder.
+1. There are variable name typos, inconsistent function names, and [confusing function responsibilities](https://en.wikipedia.org/wiki/Single-responsibility_principle).
+1. The "`checkNeighbors`" method requires more intricate tests. Due to this, and as the [SonarLint extension](https://www.sonarlint.org/vscode) suggests, the method's complexity needs reducing.
+
+![CheckNeighbors Method Complexity](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2021/10/VueCheckNeighborsComplexity.png)
+
+So, what is the next step? Is it to refactor/add TypeScript or upgrade to Vue 3? Considering that Vue 3 is natively in TypeScript, performing the transition to TypeScript first is best. That way, an upgrade to Vue 3 would be even more straightforward. It could allow for more than just following [the above guidance](https://v3.vuejs.org/guide/migration/migration-build.html), like generating a new Vue 3 project then copying over the existing Vue components.
+
+## Reading the Room
+
+What does everyone do at networking events when they initially enter the space? They scan the room for their coworkers, groups dressing/acting similarly, or easy-congregation areas like water coolers. The reason? Gaining quick familiarity with the event's ins and outs allows anyone to network with people more efficiently.
+
+The same behavior occurs when joining a project for the first time. To be a high-functioning contributor, a new individual will hit the parts of an application that tell its various functionalities (hence the look at the Vue GUI earlier). Are there tests that exist to show distinctions in functionality? What is the component or folder structure of [parent-to-child relationships](https://vuejs.org/v2/guide/instance.html)? How are the methods and variables named? Quick acclimation is the goal here.
+
+Strongly-typed languages like TypeScript allow for better readability because methods and objects are well-structured. It is all about real-time feedback while developing the working code. Perhaps even more valuable is how the types can prevent exceptions from occurring during runtime too.
```

Comment:
```suggestion
Strongly-typed languages like TypeScript allow for better readability because methods and objects are well-structured. It is all about real-time feedback while developing the working code. Perhaps even more valuable is how the types can prevent exceptions from occurring during runtime, too.
```